### PR TITLE
 Handle update, create, and delete in pipeline

### LINKED ISSUE #2: Ensure pipeline deals properly with deletes
When a resource template, ACL, or other linked data resource is deleted, remove it from the index.

~~blocked by #3~~

### LINKED ISSUE #4: Ensure pipeline deals properly with edits
When a resource template, ACL, or other linked data resource is modified, replace its entry in the index.

~~blocked by #3~~

### PR DESCRIPTION
**Pull request by mjgiarlo**
Includes:

* Handle deletes separate from updates and creates in `Pipeline#run()`
  * Add `Indexer#delete()` method to handle deletes
* Refactor `Pipeline` class to extract small methods from `#run()`
* Expand indexer unit test to cover changes made to `Indexer` class
* Expand pipeline unit test to cover changes made to `Pipeline` class
* Update integration test to cover delete case
* Rename fakes to better reflect the domain language used elsewhere in the codebase:
  * Request fakes are agent fakes
  * Indexer fakes are client fakes
* Add `#delete()` method to client fakes
* Use `jest-mock-console` to eat log output (to tidy up test suite runs)
* Make CI build fail if integration tests fail (via using `docker-compose run`)
* Make CodeClimate ignore code smells in tests and mocks
* Fix `curl` incantation in README and pointed to Sinopia notes for more examples
* Prefer `case` statements in `switch` block to be indented one level deeper

Fixes #2
Fixes #4

### CODE REVIEW

File: src/pipeline.js [hunk 1]
```diff
@@ -16,16 +16,48 @@ export default class Pipeline {
...
+          this[operation](uri)
+          break
+        // We only understand the above operation types. Log an error and keep listening.
+        default:
+          this.logger.error(`unsupported operation: ${operation}`)
+      }
     })
   }
+
+  // TODO: Figure out a more Javascript-y way to have class functions share behavior
```
Comment (jermnelson) [inline src/pipeline.js]: Maybe using a JS functional approach?

File: .eslintrc.json [hunk 2]
```diff
@@ -16,7 +16,8 @@
   "rules": {
     "indent": [
       "error",
-      2
+      2,
+      { "SwitchCase": 1 }
```
Comment (ndushay) [inline .eslintrc.json]: what does this eslint change mean?

File: __tests__/pipeline.test.js [hunk 3]
```diff
@@ -44,16 +75,59 @@ describe('Pipeline', () => {
...
+        pipeline.listener = new ListenerFake(fakeBody)
+      })
+      test('logs a debug message', () => {
+        pipeline.run()
+        expect(logSpy).toHaveBeenCalledWith(`deleting ${objectUri} from index`)
+      })
+      test('calls delete on the indexer', () => {
+        let indexerSpy = jest.spyOn(pipeline.indexer, 'delete')
+        pipeline.run()
+        expect(indexerSpy).toHaveBeenCalledWith(objectUri)
```
Comment (ndushay) [inline __tests__/pipeline.test.js]: yay!  implementing skipped tests!

File: src/pipeline.js (hunk 1 above)
Comment (mjgiarlo) [reply src/pipeline.js]: Yeah, I just don't know what that means at this point. :laughing: All of my refactoring experience and knowledge is OOP-specific, other than DRY (blech) and extracting small functions from larger functions.

File: .eslintrc.json (hunk 2 above)
Comment (mjgiarlo) [reply .eslintrc.json]: Good question, @ndushay 

Prefer this:

```javascript
switch(foo) {
  case '1':
    whatever()
  default:
    whatever()
}
```

over:

```javascript
switch(foo) {
case '1':
  whatever()
default:
  whatever()
}
```

But I acknowledge this is subject and others may in fact prefer the latter?

Comment (ndushay) [reply .eslintrc.json]: personally, I prefer the former.  So I'm good with this change.

**Pull request merged**